### DISCUSSION TITLE
put avalonia essentials to workabke state

### DISCUSSION THREAD

**Discussion by cesarchefinho** [AvaloniaUI/Avalonia#13009] [Polls]
actualy avalonia.essentials is unusable

this is a big problem because maui essentials isnt usable with avalonia too

this is THE problem! 

I don't want to have to do hacks or use hybrid solutions in my development LIKE avalonia hibrid and others. 

If I need to go to Maui to develop, why use Avalonia and not develop everything directly in Maui? Or use Uno? 

Avalonia essentials needs to be complete and robust and usable.

avalonia should be robust enough to read the camera and to display multimedia, which implies media control and camera control.

WinUi and UWP have one, WPF has it, Maui has it, Android (java) has it, IOS has it, why won't Avalonia have it? 

Will avalonia be the only framwork that doesn't handle multimedia well? 

How to justify development in Avalonia to the “bosses”?

Avalonia essentials need stay in usable state, with nuget packages, instalation istruction and basic camera end media controls.

anyone think like me ?

**Comment by MikeCodesDotNET** [comment]
Thank you for your feedback on Avalonia Essentials.

The frameworks you compare us to are backed by trillion-dollar corporations like Microsoft, Google, and Apple, which have vast resources to invest in their frameworks, which allows them to offer a wide range of features. 

Avalonia, on the other hand, is an open-source project developed by an engaged community of developers. While we (the core team) formed a company to fund working on the project full-time, we are a small team of 12 people working hard to make Avalonia as robust as possible. Our resources are limited compared to the giants of the industry. We rely heavily on contributions from our community to enhance Avalonia for the benefit of everyone. This collaborative approach has been the cornerstone of Avalonia for the last decade and is something we're incredibly proud of. 

If you find that Avalonia lacks certain features you believe are essential, I strongly encourage you to get involved. Contributing to the project is the best way to expedite adding features you need. We welcome contributions from all members of our community, as it's through this collective effort that Avalonia will continue to evolve and improve.

**Reply by cesarchefinho** [reply]
i am already involved. last month i try to contribute caractercasing to textbox

But avalonia realy need a small part of avalonia essentials (camera and multimidia)  before working on avalonia 12, if no it will fail face uno and face maui.

**Reply by MikeCodesDotNET** [reply]
I appreciate your response, but I strongly disagree that lacking camera and multimedia controls will make us fail against Uno or MAUI. Avalonia has already gained adoption from major companies like GitHub, Unity, and JetBrains, highlighting its utility beyond these specific controls. 

If these features are crucial for you, I encourage you to contribute to their development. If you're not equipped to do so, I can provide a quote for one of the core team to do the work.

**Reply by cesarchefinho** [reply]
all controls that i can develop i will submit to avalonia. like i made with textbox

but i dont have knloge to make all things that is needed to put avalonia in same level of maui.essentials and uwp/winui multimedia

**Reply by thevortexcloud** [reply]
In many cases a lot of XAML frameworks are fairly similar and you can copy and paste a lot of code (license permitting) if it's available. The hard part is ensuring compatibility with every platform Avalonia supports. But from memory (and I could be misremembering) a lot of Avalonia's code has been taken directly from WinUI and WPF.

**Reply by lylerolleman** [reply]
As much as I would also love to see a fully cross platform Avalonia.Essentials solution, it is a rather monumental task. More realistic (and one I would personally support as being a priority) is key APIs, rather than the full Essentials API. Ex. secure storage, preferences, web authenticator etc. as you'll be hard pressed to build an application that doesn't include at least a few of them. Other, more platform specific APIs can probably wait (ex. Accelerometer, as you'll probably only care about it on mobile, where you can just use MAUI.Essentials)

It's worth noting that you absolutely can use MAUI.Essentials in Avalonia on supported platforms. This can be combined with other libraries (or custom code) on the unsupported platforms as needed. Not ideal sure, but certainly possible 

The same is true of any non-control library from MAUI (and even Xamarin)

**Reply by cesarchefinho** [reply]
Definitely maui.essentials dont work with avalonia. there are a lot of shortcomings, bugs and problems.

**Reply by MikeCodesDotNET** [reply]
> there are a lot of shortcomings, bugs and problems.

You’ve said this a few times, but I don’t see any bug reports on the repo. Can you please log the bugs and problems on the [Essentials repo](https://github.com/AvaloniaUI/Avalonia.Essentials), with enough information for us to investigate. 

Avalonia.Essentials is a fork of the MAUI Essentials, so it’s surprising that you’re hitting bugs and problems as it’s a very mature SDK.
